x-pack/plugins/observability_solution/infra/server/lib/alerting/common/get_values.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const getEvaluationValues = <T extends { currentValue: number | null }>(
+  alertResults: Array<Record<string, T>>,
+  group: string
+): Array<number | null> => {
+  return alertResults.reduce((acc: Array<number | null>, result) => {
+    if (result[group]) {
+      acc.push(result[group].currentValue);
+    }
+    return acc;
+  }, []);
+};
+
+export const getThresholds = <T extends { threshold: number[] }>(
+  criteria: T[]
+): number[] | undefined => {
+  let hasMultipleThresholdComparator = false;
+  const thresholds = criteria
+    .map((c) => c.threshold)
+    .reduce((acc: number[], t) => {
+      if (t.length !== 1) {
+        hasMultipleThresholdComparator = true;
+        return acc;
+      }
+      return acc.concat(t[0]);
+    }, []);
+
+  // We don't save threshold if at least one of the comparators is a multiple threshold one such as 'in between'
```

Comment:
>what we will show in the alert table does not make sense and we will not have a 1:1 relation between evaluated value and threshold in those cases.

Can help me understand your note with an example? 
Also for the custom threshold rule we persisted the threshold in the threshold-range use cases. How there we don't have this issue? Why don't we keep it consistent across the rules?

Reply:
In this PR, I also changed this logic for the custom threshold rule, so it is consistent between rules.

Reply:
> Can help me understand your note with an example?

Suppose that we have a condition with an `in between` comparator and thresholds as between [3, 5], in that case, if we include those thresholds in the `ALERT_EVALUATION_THRESHOLD`, we will have:
- ALERT_EVALUATION_VALUES: [4]
- ALERT_EVALUATION_THRESHOLD: [3, 5]

Compare it to another case when we have two conditions, one above 3 and one above 5, then we will have:
- ALERT_EVALUATION_VALUES: [4, 7]
- ALERT_EVALUATION_THRESHOLD: [3, 5]

Here, we have a one-to-one connection between the observed value and threshold but in the previous case, we need to use 2 thresholds. To keep the use case simpler, similar to what I saw in the ES Query rule, I didn't save the threshold value in case of multiple threshold comparators.

Let me know if something is not clear.

Reply:
Btw, as @benakansara suggested, I think we need to discuss how to cover those cases as well, but I guess the cases in which the user will use one threshold comparator are more frequent, and having a threshold for those cases is a good start until we come up with a plan to cover the rest as well. (For all observability rules)

Reply:
Thanks for sharing the example. It's good having consistently across rules. Yes we need to discuss the multiple threshold. And I do remember that I raised that points in the Custom threshold PR.